File: soap/Q2213_DataCountEvenOdd.java
import java.util.Arrays;
import java.util.List;
import soap.q2211.generated.DataService;
import soap.q2211.generated.SoapDataService;

public class Q2213_DataCountEvenOdd {
    static final String STUDENT_CODE = "B21DCCN001";   // TODO
    static final String Q_CODE       = "TODO_qCode";   // TODO

    public static void main(String[] args) {
        DataService service = new DataService();
        SoapDataService port = service.getSoapDataServicePort();

        List<Integer> data = port.getData(STUDENT_CODE, Q_CODE);
        System.out.println("data=" + data);

        int even = 0, odd = 0;
        for (int n : data) {
            if (n % 2 == 0) even++; else odd++;
        }
        List<String> answer = Arrays.asList("EVEN=" + even, "ODD=" + odd);
        System.out.println("answer=" + answer);

        port.submitDataStringArray(STUDENT_CODE, Q_CODE, answer);
        System.out.println("Đã submit " + answer);
    }
}

File: soap/Q2213_DataCountEvenOdd.py
from zeep import Client

EXAM_IP      = "36.50.135.242"
STUDENT_CODE = "B22DCCN863"   # TODO
Q_CODE       = "TODO_qCode"   # TODO
WSDL         = f"http://{EXAM_IP}:2221/DataService?wsdl"


def main():
    client = Client(WSDL)

    data = client.service.getData(STUDENT_CODE, Q_CODE)
    print("data=", data)

    even = sum(1 for n in data if int(n) % 2 == 0)
    odd  = len(data) - even
    answer = [f"EVEN={even}", f"ODD={odd}"]
    print("answer=", answer)

    client.service.submitDataStringArray(STUDENT_CODE, Q_CODE, answer)
    print("Đã submit", answer)


if __name__ == "__main__":
    main()
